### PR TITLE
[SDESK-7281] Upgrade Flask to v3

### PR DESCRIPTION
**Pull request by eos87**
This PR is far from being ready to be merged. It has been only opened to allow collaboration with other team members. 

[SDESK-7281]

[SDESK-7281]: https://sofab.atlassian.net/browse/SDESK-7281?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -17,9 +17,8 @@
     "eve>=1.1.2,<=2.1.0",
     "eve-elastic>=7.4.0,<7.5.0",
     "elasticsearch[async]<7.18",  # we are using oss version on test server
-    "flask>=1.1,<1.2",
+    "flask>=3.0",
```
Comment (MarkLark86) [inline setup.py:20]: We will need to include the `async` extra for Flask.
i.e. `flask[async]>=3.0`

**Comment by MarkLark86** [comment]
@eos87 I'm okay with merging this PR when you're ready. Our current focus is Newshub (which works with this branch), and we will need more PRs against superdesk-core. Merging this will help there.

As we already have tickets to fix deprecations, we can use that ticket to make Superdesk run and tests pass etc.

File: setup.py (hunk 1 above)
Comment (eos87) [reply setup.py]: done

**Comment by eos87** [comment]
> @eos87 I'm okay with merging this PR when you're ready. Our current focus is Newshub (which works with this branch), and we will need more PRs against superdesk-core. Merging this will help there.
> 
> As we already have tickets to fix deprecations, we can use that ticket to make Superdesk run and tests pass etc.

I agree. I will merge it in a little while.

**Pull request merged**